File: python/src/unittest/python/GoogleDocTest.py
import unittest
from Documents import GoogleDoc
from BaseModule import *


class GoogleDocTest(unittest.TestCase):

    def setUp(self):
        self.google_doc = GoogleDoc()

    def test_font(self):
        font = self.google_doc.get_font()
        self.assertEqual("TimesNewRoman", font.get_font_name())
        self.assertEqual(25, font.get_font_size())

    def test_style(self):
        style = self.google_doc.get_style()
        self.assertEqual("TimesNewRoman", style.get_font().get_font_name())
        self.assertEqual(25, style.get_font().get_font_size())

    def test_background(self):
        background = self.google_doc.get_background()
        self.assertEqual(400.0, background.get_width(), 0.0)
        self.assertEqual(600.0, background.get_height(), 0.0)

    def test_sharing_permissions(self):
        self.assertEqual(2, self.google_doc.get_sharing_permissions())


if __name__ == '__main__':
    unittest.main()

File: csharp/adapter_practice/test/GoogleDocTest.cs
using adapter_practice.main.document;
using adapter_practice.main.other;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;

namespace adapter_practice.test
{
    [TestFixture]
    class GoogleDocTest
    {
        /**
 * Google document.
 */
        private GoogleDoc gd;
        [SetUp]
    public void SetUp()
        {
            gd = new GoogleDoc();
        }
        /**
         * Font test.
         */
        [TestCase]
    public void testFont()
        {
            Font font = gd.getFont();
            Assert.AreEqual("Arial", font.getName());
        }
        /**
         * Style test.
         */
        [TestCase]
        public void styleTest()
        {
            Object style = new Object();
            Assert.AreEqual(true, style.Equals(gd.getStyle()));
        }
        /**
         * Background image test.
         */
        [TestCase]
        public void backgroundImageTest()
        {
            BackgroundImage bi = new BackgroundImage();
            Assert.AreEqual(true, bi.Equals(gd.getBackground()));
        }
        /**
         * Sharing permissions test.
         */
        [TestCase]
        public void permissionsTest()
        {
            Assert.AreEqual("Sharing permissions are not enabled", gd.setSharingPermissions(0));
            Assert.AreEqual("Sharing permissions enabled", gd.setSharingPermissions(1));
        }
    }
}
